src/UI/components/speciesInformation/speciesInformationEditor.tsx

```diff
@@ -0,0 +1,109 @@
+    'initial description'
+  );
+  const [shortDescription, setShortDescription] = useState('');
+  const [name, setName] = useState('');
+  const dispatch = useAppDispatch();
+  const token = useAppSelector((s) => s.auth.token);
+  const currentSpeciesInformation = useAppSelector(
+    (s) => s.speciesInfo.currentInfoForEditing
+  );
+
+  const router = useRouter();
+  const id = router.query.id as string | undefined;
+
+  const saveSpeciesInformation = useCallback(() => {
+    const speciesInformation: SpeciesInformation = {
+      id,
+      name,
+      shortDescription,
+      description,
+    };
+    dispatch(upsertSpeciesInformation(speciesInformation));
+  }, [dispatch, id, description, name, shortDescription]);
+
+  useEffect(() => {
+    if (id) {
+      dispatch(getSpeciesInformation(id));
+    }
+  }, [dispatch, id]);
+
+  useEffect(() => {
+    if (currentSpeciesInformation) {
+      setName(currentSpeciesInformation.name);
+      setShortDescription(currentSpeciesInformation.shortDescription);
+      console.log(currentSpeciesInformation.description);
+      setDescription(currentSpeciesInformation.description);
+      setInitialDescription(currentSpeciesInformation.description);
+    }
+  }, [currentSpeciesInformation]);
+
+  return (
+    <div>
+      <Typography variant="h4" sx={{ mt: 2, mb: 1 }}>
+        {id ? 'Edit' : 'Create'} species information
+      </Typography>
+      <Typography color="primary" variant="h5" sx={{ mt: 2, mb: 1 }}>
+        Name
+      </Typography>
+      <TextField
+        variant="outlined"
+        sx={{ width: '100%' }}
+        value={name}
+        onChange={(e) => setName(e.target.value)}
+      />
+      <Typography color="primary" variant="h5" sx={{ mt: 2, mb: 1 }}>
+        Short description
+      </Typography>
+      <TextField
+        multiline
+        rows={4}
+        sx={{ width: '100%' }}
+        value={shortDescription}
+        onChange={(e) => setShortDescription(e.target.value)}
+      />
+      <Typography color="primary" variant="h5" sx={{ mt: 2, mb: 1 }}>
+        Full Description
+      </Typography>
+      <TextEditor
+        description={description}
+        setDescription={setDescription}
+        initialDescription={initialDescription}
+      />
+      <Typography color="primary" variant="h5" sx={{ mt: 2, mb: 1 }}>
+        Species image
+      </Typography>
+      <div>Placeholder</div>
+      <Typography color="primary" variant="h5" sx={{ mt: 2, mb: 1 }}>
+        Distribution map image
+      </Typography>
+      <div>Placeholder</div>
+      <div
+        style={{ display: 'flex', justifyContent: 'flex-end', marginTop: 15 }}
+      >
+        <Button
+          variant="contained"
+          onClick={saveSpeciesInformation}
+          sx={{ m: 0, minWidth: 150 }}
+        >
+          {id ? 'Update' : 'Create'}
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+export default SpeciesInformationEditor;
```

Comment:
Ooh, no tests... Intentional?

Reply:
For the moment yes, in order to get Chris going on the next story - I'll come back and add tests for this one in my next PR.